lambda/src/main/scala/pricemigrationengine/services/CohortTableLive.scala

```diff
@@ -195,23 +195,34 @@ object CohortTableLive {
       dependencies: DynamoDBZIO with StageConfiguration with CohortTableConfiguration with Logging =>
         new Service {
           override def fetch(
-              filter: CohortTableFilter
+            filter: CohortTableFilter,
+            latestStartDateInclusive: Option[LocalDate]
           ): IO[CohortFetchFailure, ZStream[Any, CohortFetchFailure, CohortItem]] = {
             for {
               cohortTableConfig <- CohortTableConfiguration.cohortTableConfig
                 .mapError(error => CohortFetchFailure(s"Failed to get configuration:${error.reason}"))
               stageConfig <- StageConfiguration.stageConfig
                 .mapError(error => CohortFetchFailure(s"Failed to get configuration:${error.reason}"))
+              queryRequest = new QueryRequest()
+                .withTableName(s"PriceMigrationEngine${stageConfig.stage}")
+                .withIndexName("ProcessingStageIndexV3")
+                .withKeyConditionExpression(
+                  "processingStage = :processingStage" + latestStartDateInclusive.fold("") { _ =>
+                    " AND startDate <= :latestStartDateInclusive"
```

Comment:
I thought we have to give *at least* 30 days' notice so shouldn't this be sending out `>=`?

Reply:
If I understand your point correctly, that wasn't my understanding of the requirements.

I had thought we would run the job every day, and send out notifications for items that had a start date in 30 days. The 'less than' is just there to make sure we hoover up any items that were missed for any reason.

If there are strict rules around the dm having to be send out no less than 30 days ahead of the start date, we need to take into account details Latcham lead time and batching behaviour. We can do that once we know what they are.

Ill double check with Sabrina what the rules are.

Reply:
👍 
In that case this is good to go

Reply:
Yes so I checked with Sabrina and I think this is the correct behaviour, pending some tweaking around with the lead times.